apps/patient-listener/src/main/java/gov/cdc/nbs/patientlistener/request/update/PatientUpdateRequestHandler.java

```diff
@@ -104,6 +108,89 @@ public void handlePatientSexAndBirthUpdate(final UpdateSexAndBirthData data) {
     );
   }
 
+  /**
+   * Sets the Person's administrative info to the specified values.
+   *
+   * @param data
+   */
+  @Transactional
+  public void handlePatientAdministrativeUpdate(final UpdateAdministrativeData data) {
+    if (!userService.isAuthorized(data.updatedBy(), VIEW_PATIENT, EDIT_PATIENT)) {
+      throw new UserNotAuthorizedException(data.requestId());
+    }
+
+    var person = findPerson(data.patientId(), data.requestId());
+    patientUpdater.update(person, data);
+    updateElasticsearchPatient(person);
+    statusProducer.successful(
+        data.requestId(),
+        "Successfully updated administrative info",
+        data.patientId()
+    );
+  }
+
+    /**
+   * Add a person name.
+   *
+   * @param data
+   */
+  @Transactional
+  public void handlePatientNameAdd(final AddNameData data) {
+    if (!userService.isAuthorized(data.updatedBy(), VIEW_PATIENT, EDIT_PATIENT)) {
+      throw new UserNotAuthorizedException(data.requestId());
+    }
+
+    var person = findPerson(data.patientId(), data.requestId());
+    patientUpdater.update(person, data);
+    updateElasticsearchPatient(person);
+    statusProducer.successful(
+        data.requestId(),
+        "Successfully add a patient name",
+        data.patientId()
+    );
+  }
+
+  /**
+   * Update a person name.
+   *
+   * @param data
+   */
+  @Transactional
+  public void handlePatientNameUpdate(final UpdateNameData data) {
+    if (!userService.isAuthorized(data.updatedBy(), VIEW_PATIENT, EDIT_PATIENT)) {
+      throw new UserNotAuthorizedException(data.requestId());
+    }
+
+    var person = findPerson(data.patientId(), data.requestId());
+    patientUpdater.update(person, data);
+    updateElasticsearchPatient(person);
+    statusProducer.successful(
+        data.requestId(),
+        "Successfully updated patient name",
+        data.patientId()
+    );
+  }
+
+    /**
+   * Delete a person name.
+   *
+   * @param data
+   */
+  @Transactional
+  public void handlePatientNameDelete(final String requestId, final long patientId, short personNameSeq, final long userId) {
+    if (!userService.isAuthorized(userId, VIEW_PATIENT, EDIT_PATIENT)) {
+      throw new UserNotAuthorizedException(requestId);
+    }
+    var person = findPerson(patientId, requestId);
+    patientUpdater.update(person, new DeleteNameData(patientId, personNameSeq, requestId, userId, null));
+    updateElasticsearchPatient(person);
+    statusProducer.successful(
+        requestId,
+        "Successfully deleted patient name",
+        userId
+    );
+  }
+
```

Comment:
I don't believe there is any test coverage for these new methods.

Reply:
ah i thought coverage calcs were wrong! good catch...